lib/transproc/transformer/class_interface.rb

```diff
@@ -53,6 +53,33 @@ def container(container = ::Transproc::Undefined)
         end
       end
 
+      # Define an anonymous transproc derived from given Transformer
+      # Evaluates block with transformations and returns initialized transproc.
+      # Does not mutate original Transformer
+      #
+      # @example
+      #
+      #   class MyTransformer < Transproc::Transformer[MyContainer]
+      #   end
+      #
+      #   transproc = MyTransformer.define do
+      #     map_values t(:to_string)
+      #   end
+      #   transproc.call(a: 1, b: 2)
+      #   # => {a: '1', b: '2'}
+      #
+      # @yield Block allowing to define transformations. The same as class level DSL
+      #
+      # @return [Function] Composed transproc
+      #
+      # @api public
+      def define(&block)
+        return transproc unless block_given?
+
+        Class.new(self).tap { |klass| klass.instance_eval(&block) }.transproc
+      end
+      alias build define
```

Comment:
ROM Mapper has `#build` for initialization. I thought it doesn't hurt to add this. 
`#build` makes more sense without block while `#define` with block. 
In future we may refactor instance-level and `#new` will sounds weird